aten/src/ATen/TensorIndexing.h

```diff
@@ -37,52 +37,47 @@ TORCH_API extern const EllipsisIndexType Ellipsis;
 
 struct TORCH_API Slice final {
  public:
-  // This mirrors `__PySlice_Unpack` in torch/csrc/utils/python_compat.h
   Slice(
-      c10::optional<int64_t> start_index = c10::nullopt,
-      c10::optional<int64_t> stop_index = c10::nullopt,
-      c10::optional<int64_t> step_index = c10::nullopt) {
+      c10::optional<c10::SymInt> start_index = c10::nullopt,
+      c10::optional<c10::SymInt> stop_index = c10::nullopt,
+      c10::optional<c10::SymInt> step_index = c10::nullopt) {
     if (!step_index.has_value()) {
-      step_ = 1;
+      step_ = c10::SymInt(1);
     } else {
       step_ = step_index.value();
-      TORCH_CHECK_VALUE(step_ != 0, "slice step cannot be zero");
-
-      // Here step might be -INDEX_MAX-1; in this case we replace it
-      // with -INDEX_MAX.  This doesn't affect the semantics, and it
-      // guards against later undefined behaviour resulting from code that
-      // does "step = -step" as part of a slice reversal.
-      if (step_ < -INDEX_MAX)
-        step_ = -INDEX_MAX;
     }
+
+    TORCH_CHECK_VALUE(step_ != 0, "slice step cannot be zero");
+
     if (!start_index.has_value()) {
-      start_ = step_ < 0 ? INDEX_MAX : 0;
+      start_ = c10::SymInt(step_ < 0 ? INDEX_MAX : 0);
     } else {
       start_ = start_index.value();
     }
+
     if (!stop_index.has_value()) {
-      stop_ = step_ < 0 ? INDEX_MIN : INDEX_MAX;
+      stop_ = c10::SymInt(step_ < 0 ? INDEX_MIN : INDEX_MAX);
     } else {
       stop_ = stop_index.value();
```

Comment:
```suggestion
      stop_ = std::move(stop_index).value();
```
You can also put the std::move after the value() call, but this seems more readable to me. No need to add unnecessary trivial copies with these assignments.